### ISSUE TITLE
well it's not good news :(

### ISSUE THREAD

**Issue by glycerine** [danmux/bouncy#2]
Hey Dan @danmux 

Since you asked...

Yeah. So. Not looking good.

FoundationDB is looking about 10x slower than LevelDB. And about 4x slower than BoltDB. I ran FoundationDB on a single node, so there was no network replication overhead either.  The storage mode was memory, which is supposed to be the fastest on FoundationDB (though the docs say it still writes to disk).

Sigh.

Oh well. It was worth a look.

My fork. Updated to use modules and the latest bleve --

https://github.com/glycerine/bleve_fdb

The bleve benchmarks, updated as well --

https://github.com/glycerine/bleve-bench

**Comment by glycerine** [comment]
Ah, I see the benchmark was doing the first 50 loads sequentially. Then a batch of 50, for each subsequent count of 100. 

So this is not really measuring batched writes. And FoundationDB is known to need batching to do stuff well; with serialized writes (it has high latency), it is for sure going to look bad.

**Comment by glycerine** [comment]
No luck.

I modified the benchmark to only do batches. ( https://github.com/glycerine/bleve_fdb/blob/master/cmd/bleve-bench-fdb/index.go#L178 )

FoundationDB is still looking sad:
~~~
foundationdb 127.0s (10.4x)
boltdb       22.9s  (1.9x)
leveldb      12.2s  (1x)
~~~
Also posted on https://github.com/glycerine/bleve-bench

**Comment by glycerine** [comment]
Well after studying the FoundationDB docs a bit more, I realized that it is the number of connections to the database that is the key to getting concurrency and hiding latency. I did some studies without involving any of the bleve code, just on writes. When using 20 - 50 separate connections, I'm able to get reasonable write throughput out of FoundationDB.

For instance:
~~~
The first 3000 wikipedia articles in the bleve-bench benchmark yield:

keyCount = 2511847
smallestKey = 1
largestKey = 198
smallestValue = 1
largestValue = 40379

writing these keys to levelDB:
elap = 1.838759281s; read back 1366055.38 rec/sec
val mean = 13.189000365070374; val sd = 306.09962864883784
key mean = 22.869594366216784; key sd = 8.581939345323846

reading from levelDB, writing into FoundationDB:
50 fdb connectons; write elap = 22.670881985s; writing to fdb at 110796.17 keys/sec
total keys written = 2511847
nConn = 50;  ngoro per conn = 1; batch size = 40
~~~
It took about 11 seconds to write into levelDB.

It took about 22 seconds to write those keys into FoundationDB, using 50 database connections. Seems acceptable.

Not sure how to get multiple connections out of bleve yet. Might have to ponder it.

Closing this. Its title is misleading. :)
